src/exceptions.rs

```diff
@@ -1079,7 +1080,11 @@ mod tests {
         let invalid_utf8 = b"fo\xd8o";
         #[cfg_attr(invalid_from_utf8_lint, allow(invalid_from_utf8))]
         let err = std::str::from_utf8(invalid_utf8).expect_err("should be invalid utf8");
-        PyErr::from_value(PyUnicodeDecodeError::new_utf8(py, invalid_utf8, err).unwrap())
+        PyErr::from_value_bound(
+            &PyUnicodeDecodeError::new_utf8(py, invalid_utf8, err)
```

Comment:
Ah, I guess we'll need to add bound constructors for `PyUnicodeDecodeError`, I added this to #3684

Reply:
I think this is a blocker to moving to the version of `value_bound` that takes `Bound` instead of `&Bound`.